File: C_Programming/Practise_Programs/PP_171.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_171
//
//  Description       : This program accepts an integer number and a
//                      bit position from the user and toggles the
//                      specified bit using a separate function.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 21/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UINT;

//////////////////////////////////////////////////////////////////
//
//  Function Name     : ToggleBit()
//
//  Description       : It is used to toggle the bit at the specified
//                      position of the given number using the
//                      bitwise XOR operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 21/08/2026
//
//////////////////////////////////////////////////////////////////

UINT ToggleBit(UINT iNo, UINT iPos)
{
    UINT iMask = 0x1;
    UINT iResult = 0;

    iMask = iMask << (iPos - 1);

    iResult = iNo ^ iMask;

    return iResult;
}

int main()
{
    UINT iValue = 0;
    UINT iRet = 0;
    UINT iLocation = 0;

    printf("Enter Number : \n");
    scanf("%u", &iValue);

    printf("Enter Bit Position : \n");
    scanf("%u", &iLocation);    
    
    iRet = ToggleBit(iValue, iLocation);

    printf("Updated number is : %u\n", iRet);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      10
//      4
//
//  Sample Output :
//
//      Enter Number :
//      Enter Bit Position :
//      Updated number is : 2
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_171.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_171
//
//  Description       : This program accepts a number from the user
//                      and displays a right-angled triangular pattern
//                      using star (*) symbols.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 05/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner; 
 
class Pattern 
{ 
    public static void Display(int iNo) 
    { 
        int i, j = 0; 
 
        for(i = 1; i <= iNo; i++) 
        { 
            for(j = 1; j <= i; j++) 
            { 
                System.out.print("*\t"); 
            } 
            System.out.println(); 
        } 
    } 
} 
 
class PP_171 
{ 
    public static void main(String A[]) 
    { 
        Scanner sobj = new Scanner(System.in); 
 
        int iValue = 0; 
 
        System.out.println("Enter the number : "); 
        iValue = sobj.nextInt(); 
 
        Pattern pobj = new Pattern(); 
 
        pobj.Display(iValue); 
    } 
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the number :
//      5
//
//  Sample Output :
//      *
//      *   *
//      *   *   *
//      *   *   *   *
//      *   *   *   *   *
//
//////////////////////////////////////////////////////////////////